components/service/experiments/src/test/java/mozilla/components/service/experiments/ActiveExperimentTest.kt

```diff
@@ -0,0 +1,144 @@
+import org.junit.runner.RunWith
+import org.mockito.ArgumentMatchers.eq
+import org.mockito.ArgumentMatchers.anyString
+import org.mockito.Mockito.`when`
+import org.mockito.Mockito.times
+import org.mockito.Mockito.verify
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class ActiveExperimentTest {
+    private lateinit var context: Context
+    private lateinit var sharedPrefs: SharedPreferences
+    private lateinit var prefsEditor: SharedPreferences.Editor
+    private val prefNameActiveExperiment = ActiveExperiment.PREF_NAME_ACTIVE_EXPERIMENT
+    private val prefKeyId = ActiveExperiment.KEY_ID
+    private val prefKeyBranch = ActiveExperiment.KEY_BRANCH
+
+    private fun testReset() {
+        context = mock()
+        sharedPrefs = mock()
+        prefsEditor = mock()
+
+        `when`(context.getSharedPreferences(eq(prefNameActiveExperiment), eq(Context.MODE_PRIVATE))).thenReturn(sharedPrefs)
+        `when`(sharedPrefs.edit()).thenReturn(prefsEditor)
+        `when`(prefsEditor.putString(anyString(), anyString())).thenReturn(prefsEditor)
+        `when`(prefsEditor.clear()).thenReturn(prefsEditor)
+    }
+
+    private fun setActiveExperimentPrefs(id: String, branch: String) {
+        `when`(sharedPrefs.getString(eq(prefKeyId), eq(null))).thenReturn(id)
+        `when`(sharedPrefs.getString(eq(prefKeyBranch), eq(null))).thenReturn(branch)
+    }
+
+    @Test
+    fun `test basics`() {
+        val experiment = testDefaultExperiment(
+            id = "id",
+            branches = listOf(
+                Experiment.Branch("branch1", 1),
+                Experiment.Branch("branch2", 2)
+            )
+        )
+
+        val activeExperiment = ActiveExperiment(experiment, "branch2")
+        assertEquals(activeExperiment.experiment, experiment)
+        assertEquals(activeExperiment.branch, "branch2")
+    }
+
+    @Test
+    fun `test load from storage`() {
+        testReset()
+        val experiments = listOf(
+            testDefaultExperiment(
+                id = "id1",
+                branches = listOf(
+                    Experiment.Branch("branch1", 2),
+                    Experiment.Branch("branch2", 3)
+                )
+            ),
+            testDefaultExperiment(
+                id = "id2",
+                branches = listOf(
+                    Experiment.Branch("branch3", 2)
+                )
+            )
+        )
+
+        // Start with empty storage.
+        assertNull(ActiveExperiment.load(context, experiments))
+        verify(context).getSharedPreferences(prefNameActiveExperiment, Context.MODE_PRIVATE)
+        verify(sharedPrefs).getString(prefKeyId, null)
+
+        // Test valid experiment id and branch name variations.
+        setActiveExperimentPrefs("id1", "branch1")
+        var active = ActiveExperiment.load(context, experiments)
+        assertEquals(experiments[0], active!!.experiment)
+        assertEquals("branch1", active!!.branch)
```

Comment:
nit: `!!` isn't needed on this line because the previous line's `!!` would have exceptioned and Kotlin is smart enough to figure this out.